xclim/ensembles/_base.py

```diff
@@ -125,12 +129,18 @@ def ensemble_mean_std_max_min(ens: xr.Dataset) -> xr.Dataset:
     """
     ds_out = xr.Dataset(attrs=ens.attrs)
     for v in ens.data_vars:
-
-        ds_out[f"{v}_mean"] = ens[v].mean(dim="realization")
-        ds_out[f"{v}_stdev"] = ens[v].std(dim="realization")
+        if weights is None:
+            ds_out[f"{v}_mean"] = ens[v].mean(dim="realization")
+            ds_out[f"{v}_stdev"] = ens[v].std(dim="realization")
+        else:
+            ds_out[f"{v}_mean"] = ens[v].weighted(weights).mean(dim="realization")
+            ds_out[f"{v}_stdev"] = ens[v].weighted(weights).std(dim="realization")
         ds_out[f"{v}_max"] = ens[v].max(dim="realization")
         ds_out[f"{v}_min"] = ens[v].min(dim="realization")
-        for vv in ds_out.data_vars:
+        for vv in [
+            m + str(n)
+            for m, n in zip([f"{v}"] * 4, ["_mean", "_stdev", "_max", "_min"])
+        ]:
```

Comment:
```suggestion
        for stat in ["mean", "stdev", "max", "min"]:
            vv = f"{v}_{stat}"
```
This change would allow a simplification of the description formatting below, but I can't suggest changes on non-modified lines.